shared/chat/conversation/messages/message-popup/exploding/index.tsx

```diff
@@ -72,57 +63,73 @@ class ExplodingPopupHeader extends React.Component<Props, State> {
   }
 
   render() {
-    const {author, deviceName, deviceRevokedAt, hideTimer, timestamp, yourMessage} = this.props
+    const {author, botUsername, deviceName, deviceRevokedAt, hideTimer, timestamp, yourMessage} = this.props
     const whoRevoked = yourMessage ? 'You' : author
     return (
-      <Box2 direction="vertical" fullWidth={true} style={styles.popupContainer}>
-        <Icon style={styles.headerIcon} type={headerIconType} />
-        <Box2 direction="vertical" style={styles.messageInfoContainer}>
-          <Box2 direction="vertical">
-            <Text type="BodySmall" style={{color: Styles.globalColors.black}}>
+      <Kb.Box2 direction="vertical" fullWidth={true} style={styles.popupContainer}>
+        <Kb.Icon style={styles.headerIcon} type={headerIconType} />
+        <Kb.Box2 direction="vertical" style={styles.messageInfoContainer}>
+          <Kb.Box2 direction="vertical">
+            <Kb.Text type="BodySmall" style={{color: Styles.globalColors.black}}>
               EXPLODING MESSAGE
-            </Text>
-          </Box2>
-          <Box2 direction="horizontal">
-            <Text type="BodySmall">by</Text>
-            <Box2 direction="horizontal" gap="xtiny" gapStart={true} style={styles.user}>
-              <Avatar username={author} size={16} onClick="profile" />
-              <ConnectedUsernames
+            </Kb.Text>
+          </Kb.Box2>
+          <Kb.Box2 direction="horizontal">
+            <Kb.Text type="BodySmall">by</Kb.Text>
+            <Kb.Box2 direction="horizontal" gap="xtiny" gapStart={true} style={styles.user}>
+              <Kb.Avatar username={author} size={16} onClick="profile" />
+              <Kb.ConnectedUsernames
                 onUsernameClicked="profile"
                 colorFollowing={true}
                 colorYou={true}
                 usernames={[author]}
                 underline={true}
                 type="BodySmallSemibold"
               />
-            </Box2>
-          </Box2>
-          <Box2 direction="horizontal">
-            <Text center={true} type="BodySmall">
+            </Kb.Box2>
+          </Kb.Box2>
+          <Kb.Box2 direction="horizontal">
+            <Kb.Text center={true} type="BodySmall">
               from device {deviceName}
-            </Text>
-          </Box2>
-          <Box2 direction="horizontal">
-            <Text center={true} type="BodySmall">
+            </Kb.Text>
+          </Kb.Box2>
+          {botUsername && (
+            <Kb.Box2 direction="horizontal">
```

Comment:
might be useful to have a component that is this avatar-with-username thing. I know we have a vertical one.